src/commands/getRunExec.ts

```diff
@@ -14,7 +13,9 @@ class RunExecStruct implements CommandExecStruct {
 	argsNeedDoubleDashes: boolean;
 
 	constructor(
-		packageManager: Pick<PackageManager, 'name' | 'version' | 'getPackageInfo'>,
+		packageManager: Pick<PackageManager, 'name' | 'version' | 'getPackageInfo'> & {
+			metadata: Pick<PackageManagerMetadata, 'isYarnClassic'>;
+		},
```

Comment:
just a thought, but could we pull this type out and give it a name so it can be reused a little further down, since it's getting kinda beefy now?

Reply:
yeah a totally valid point, does [my last commit](https://github.com/james-elicx/package-manager-manager/pull/44/commits/f06688c3532ef3a571bf69d9b4391902f72a559a) look good to you? :slightly_smiling_face:

Reply:
yes thank you 😍